### PR TITLE
add pending state to annotation controller pod monitoring

### PR DESCRIPTION
**Pull request by achrefbensaad**

### CODE REVIEW

**Review by seswarrajan — APPROVED** [review]

File: pkg/KubeArmorAnnotation/controllers/pod-refresh.go [hunk 1]
```diff
@@ -60,7 +57,7 @@ func (r *PodRefresherReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			}
 		}
 	}
-
+	time.Sleep(30 * time.Second)
```
Comment (nyrahul) [inline pkg/KubeArmorAnnotation/controllers/pod-refresh.go:60]: Why this 30 sec sleep?

File: pkg/KubeArmorAnnotation/controllers/pod-refresh.go [hunk 2]
```diff
@@ -24,17 +24,14 @@ type PodRefresherReconciler struct {
...
 	var podList corev1.PodList
 	if err := r.List(ctx, &podList); err != nil {
 		log.Error(err, "Unable to list pods")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-
+	log.Info("Watching for blocked pods")
 	for _, pod := range podList.Items {
-		if pod.Status.Phase == corev1.PodRunning && strings.Contains(pod.Status.Message, "Cannot enforce AppArmor") {
+		if (pod.Status.Phase == corev1.PodRunning || pod.Status.Phase == corev1.PodPending) && strings.Contains(pod.Status.Message, "Cannot enforce AppArmor") {
```
Comment (nyrahul) [inline pkg/KubeArmorAnnotation/controllers/pod-refresh.go]: Why are we looking for pods in certain phase only for the given Message? Why can't we restart the pods regardless of the phase with the given message?

Comment (achrefbensaad) [reply pkg/KubeArmorAnnotation/controllers/pod-refresh.go]: Done

File: pkg/KubeArmorAnnotation/controllers/pod-refresh.go (hunk 1 above)
Comment (achrefbensaad) [reply pkg/KubeArmorAnnotation/controllers/pod-refresh.go:60]: To give time for the apparmor profile to be loaded.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**